src/UiPath.Workflow.Runtime/Statements/Delay.cs

```diff
@@ -76,6 +85,16 @@ protected override void Abort(NativeActivityAbortContext context)
         base.Abort(context);
     }
 
+    private void OnTimerFired(NativeActivityContext context, Bookmark bookmark, object value)
```

Comment:
afaik we don't need to call exit, as the variable goes out of scope and the NoPersistHandle disappears.